### PR TITLE
Add showname and selfname colors to courtroom_fonts.ini

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Cyan = selfname, light gray = other name
image

image

To implement, in `courtroom_fonts.ini`:
```ini
ic_chatlog_selfname_color = 0, 255, 255
ic_chatlog_showname_color = 200, 200, 200
```

Closes https://github.com/AttorneyOnline/AO2-Client/issues/671
Also overhauls `chatlogpiece.cpp` and `chatlogpiece.h` slightly
TODO: a settings option to ignore selfname

### CODE REVIEW

**Comment by Crystalwarrior** [comment]
if anyone can make a settings option for me that'd be amazing <3

File: include/chatlogpiece.h [hunk 1]
```diff
@@ -10,24 +10,26 @@ class chatlogpiece {
...
-  int get_chat_color();
+  QString get_name() { return name; };
+  QString get_showname() { return showname; };
+  QString get_message() { return message; };
+  QString get_action() { return action; };
+  bool get_selfname() { return selfname; };
+  QDateTime get_datetime() { return datetime; };
+  QString get_datetime_as_string() { return datetime.toString(); };
+  int get_chat_color() { return color; };
   QString get_full();
```
Comment (oldmud0) [inline include/chatlogpiece.h:25]:
```suggestion
  QString &get_name() const { return name; };
  QString &get_showname() const { return showname; };
  QString &get_message() const { return message; };
  QString &get_action() const { return action; };
  bool get_selfname() const { return selfname; };
  QDateTime &get_datetime() const { return datetime; };
  QString &get_datetime_as_string() const { return datetime.toString(); };
  int get_chat_color() const { return color; };
  QString &get_full() const;
```

However, getters are mostly pointless in this case unless you intend to fire events from them, which you are not. So, they are pointless. This seems to be more of a forceful way of declaring the class immutable, but you can already do that with `const` in C++ (it is strong immutability as opposed to the "fake" immutability you see in other languages).

File: src/courtroom.cpp [hunk 2]
```diff
@@ -3151,7 +3159,10 @@ void Courtroom::append_ic_text(QString p_text, QString p_name, QString p_action,
   }
 
   // Format the name of the actor
-  ui_ic_chatlog->textCursor().insertText(p_name, bold);
+  if (selfname)
+    ui_ic_chatlog->textCursor().insertText(p_name, own_name);
+  else
+    ui_ic_chatlog->textCursor().insertText(p_name, other_name);
```
Comment (oldmud0) [inline src/courtroom.cpp:3165]:
```suggestion
  QTextCharFormat name_format = selfname ? own_name : other_name;
  ui_ic_chatlog->textCursor().insertText(p_name, name_format);
```

**Review by oldmud0 — COMMENTED** [review]
It's ok. Just a lot of boilerplate I guess.

File: src/courtroom.cpp [hunk 3]
```diff
@@ -3862,13 +3871,14 @@ void Courtroom::handle_song(QStringList *p_contents)
       }
     }
     if (!mute_map.value(n_char)) {
+      bool selfname = n_char == m_cid;
       if (is_stop) {
-        log_ic_text(str_char, str_show, "", tr("has stopped the music"));
-        append_ic_text("", str_show, tr("has stopped the music"));
+        log_ic_text(str_char, str_show, "", tr("has stopped the music"), 0, selfname);
+        append_ic_text("", str_show, tr("has stopped the music"), 0, selfname);
```
[the remark this answers is not in the document]
Comment (Crystalwarrior) [reply src/courtroom.cpp:3877]: wtf? where is it seeing "5"

File: include/chatlogpiece.h [hunk 4]
```diff
@@ -10,24 +10,26 @@ class chatlogpiece {
...
-  QString get_datetime_as_string();
-  int get_chat_color();
+  QString get_name() { return name; };
+  QString get_showname() { return showname; };
+  QString get_message() { return message; };
+  QString get_action() { return action; };
+  bool get_selfname() const { return selfname; };
+  QDateTime get_datetime() { return datetime; };
+  QString get_datetime_as_string() { return datetime.toString(); };
+  int get_chat_color() const { return color; };
```
[the remark this answers is not in the document]
Comment (Crystalwarrior) [reply include/chatlogpiece.h:24]: wtf u smoking

File: src/courtroom.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (oldmud0) [reply src/courtroom.cpp:3877]: 5

**Review by oldmud0 — APPROVED** [review]
we should add `ic_chatlog_selfname_color`, `ic_chatlog_showname_color` to courtroom_fonts for the default theme, even if we're not actually changing the colors

**Pull request merged**